### PR TITLE
LPS-203826 The URL of content on a display page redirects to the login page when the Guest user does not have permissions

### PR DESCRIPTION
**Pull request by lfbesada**
https://liferay.atlassian.net/browse/LPS-203826

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by lfbesada** [comment]
ci:test:sf

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java [hunk 1]
```diff
@@ -124,15 +260,35 @@ private void _assertIncludeLayoutContent(
...
+				new MockHttpServletResponse();
+
 			layoutTypeController.includeLayoutContent(
-				_getMockHttpServletRequest(layout, user),
-				new MockHttpServletResponse(), layout);
+				mockHttpServletRequest, mockHttpServletResponse, layout);
 
 			Assert.assertFalse(noSuchLayoutExceptionExpected);
+
+			Assert.assertTrue(
```
Comment (ealonso) [inline modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:282]: @lfbesada here you can user Assert.assertEquals

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java [hunk 2]
```diff
@@ -107,10 +198,55 @@ public void testDisplayPageTypeControllerWithoutContextInfoItem()
 		_assertIncludeLayoutContent(
 			false, layout.getPlid(), TestPropsValues.getUser());
 
-		User guestUser = _userLocalService.getGuestUser(_group.getCompanyId());
+		_assertIncludeLayoutContent(true, draftLayout.getPlid(), _guestUser);
+		_assertIncludeLayoutContent(true, layout.getPlid(), _guestUser);
+	}
+
+	private FragmentEntryLink _addFragmentEntryLink(Layout layout)
```
Comment (ealonso) [inline modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:205]: @lfbesada the return value is never used

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java [hunk 3]
```diff
@@ -107,10 +198,55 @@ public void testDisplayPageTypeControllerWithoutContextInfoItem()
...
+				).toString(),
+				fragmentEntry.getCss(), fragmentEntry.getConfiguration(),
+				fragmentEntry.getFragmentEntryId(), fragmentEntry.getHtml(),
+				fragmentEntry.getJs(), layout,
+				fragmentEntry.getFragmentEntryKey(), fragmentEntry.getType(),
+				null, 0,
+				_segmentsExperienceLocalService.
+					fetchDefaultSegmentsExperienceId(layout.getPlid()));
+
+		for (FragmentEntryLinkListener fragmentEntryLinkListener :
```
Comment (ealonso) [inline modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:242]: @lfbesada Why it is necessary?

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java [hunk 4]
```diff
@@ -178,23 +348,101 @@ private MockHttpServletRequest _getMockHttpServletRequest(
...
+		serviceContext.setAddGuestPermissions(addGuestPermissions);
+
+		AssetVocabulary assetVocabulary =
+			_assetVocabularyLocalService.addVocabulary(
+				TestPropsValues.getUserId(), _group.getGroupId(),
+				RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+				RandomTestUtil.randomLocaleStringMap(), Collections.emptyMap(),
+				StringPool.BLANK, serviceContext);
+
+		_assetCategory = _assetCategoryLocalService.addCategory(
```
Comment (ealonso) [inline modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:364]: @lfbesada We are adding two asset categories, and assigning them to the same variable, is it necessary?

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java [hunk 5]
```diff
@@ -107,10 +198,55 @@ public void testDisplayPageTypeControllerWithoutContextInfoItem()
 		_assertIncludeLayoutContent(
 			false, layout.getPlid(), TestPropsValues.getUser());
```
Comment (ealonso) [inline modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:200]: @lfbesada I think we can remove the empty line

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java (hunk 3 above)
Comment (lfbesada) [reply modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:242]: I think you are right, It isn´t, I am going to remove it, thanks!

File: modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java (hunk 4 above)
Comment (lfbesada) [reply modules/apps/layout/layout-type-controller/layout-type-controller-test/src/testIntegration/java/com/liferay/layout/type/controller/display/page/internal/layout/type/controller/test/DisplayPageLayoutTypeControllerTest.java:364]: :facepalm:

**Comment by lfbesada** [comment]
@ealonso Thanks for the review :heart:  I have added your feedback, thanks

**Comment by ealonso** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by Tim-Cao** [comment]
ci:forward:force

**Pull request closed without merging**